packages/react-components/src/components/Modal/Modal.tsx

```diff
@@ -1,27 +1,22 @@
 import * as React from 'react';
 import cx from 'clsx';
 import { ModalBaseProps, ModalBase } from './ModalBase';
-import { Heading } from '../Typography';
+import { ModalHeader } from './ModalHeader';
+import { ModalFooter } from './ModalFooter';
+import { ModalCloseButton } from './ModalCloseButton';
 
 import styles from './Modal.module.scss';
-import { ModalCloseButton } from './ModalCloseButton';
 
 export interface ModalProps extends ModalBaseProps {
-  heading?: React.ReactNode;
-  labelHeading?: React.ReactNode;
   fullSpaceContent?: boolean;
-  footer?: React.ReactNode;
```

Comment:
I'm afraid that we can't just remove these props from the interface and put in the new version.
We should consider to keep it, and also use atoms in the component `children`.